### PR TITLE
enhancement: two additional events for new user analytics funnel

### PR DESCRIPTION
**Pull request by Antivortex**
Two additional events will help analytics team understand new user experience better:

avatar_edit_success_nux - triggered when user finishes editing his avatar for the first time after creating a new user (but not after other edits in the middle of playing)

terms_of_service_success_nux - triggered when user accepts terms of service during new user creation process

### CODE REVIEW

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
Instead of having a static class for `NewUserExperienceAnalytics`, we could declare it as a normal class that implements an `INewUserExperienceAnalytics` interface and then inject it as dependency into `AvatarEditorHUDController`. This will ease the implementation of unit tests and fit better with the use of SOLID principles we try to apply in our project.

You can take a look to the `SocialAnalytics` class that is following the same architecture: It implements a `ISocialAnalytics` interface that is injected in the consumers.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs [hunk 1]
```diff
@@ -746,8 +747,12 @@ public void SaveAvatar(Texture2D face256Snapshot, Texture2D bodySnapshot)
 
         WebInterface.SendSaveAvatar(avatarModel, face256Snapshot, bodySnapshot, DataStore.i.common.isSignUpFlow.Get());
         userProfile.OverrideAvatar(avatarModel, face256Snapshot);
+        
         if (DataStore.i.common.isSignUpFlow.Get())
+        {
             DataStore.i.HUDs.signupVisible.Set(true);
+            NewUserExperienceAnalytics.AvatarEditSuccessNux();
```
Comment (aixaCode) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs:754]: In this script we already have `internal IAnalytics analytics` variable. Do you think we need NewUserExperienceAnalytics class at all?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SignupHUD/SignupHUDController.cs [hunk 2]
```diff
@@ -59,6 +59,7 @@ internal void OnTermsOfServiceAgreed()
             WebInterface.SendPassport(name, email);
             DataStore.i.common.isSignUpFlow.Set(false);
             signupVisible.Set(false);
+            NewUserExperienceAnalytics.SendTermsOfServiceAcceptedNux();
```
Comment (aixaCode) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SignupHUD/SignupHUDController.cs:62]: Here we could also just pass `IAnalytics` interface.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs [hunk 3]
```diff
@@ -6,6 +6,7 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using JetBrains.Annotations;
```
Comment (aixaCode) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs:9]: This looks like got included by accident :) I don't see any usage of that namespace

**Review by kwspgoliasz — APPROVED** [review]
The feature appears to be implemented correctly, as the changes had no negative impact on the game.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs (hunk 3 above)
Comment (Antivortex) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs:9]: Yeah, this is true

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs (hunk 1 above)
Comment (Antivortex) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs]: Yes, I think it makes sense to have a specialized wrapper for it same as we do in other places i.e. SocialAnalytics or GenericAnalytics. This helps to easily find all of the events related to a particular feature scope.

**Review by sandrade-dcl — APPROVED** [review]

**Pull request merged**